### PR TITLE
OBJLoader now throws on invalid obj files

### PR DESCRIPTION
**Pull request by vidstige**
See further discussion on the solution in #7803.

### CODE REVIEW

**Comment by vidstige** [comment]
Superseding #7803. Please take a look @mrdoob and others.

File: examples/js/loaders/OBJLoader.js [hunk 1]
```diff
@@ -2,6 +2,11 @@
  * @author mrdoob / http://mrdoob.com/
  */
 
+THREE.ParseException = function(message) {
```
Comment (mrdoob) [inline examples/js/loaders/OBJLoader.js]: Uh? What's this class for?

Comment (vidstige) [inline examples/js/loaders/OBJLoader.js]: It's thrown later when an invalid line is encountered. It's so you can catch it and use the `message` property on it. Very convenient.

Comment (mrdoob) [inline examples/js/loaders/OBJLoader.js]: Why not doing this instead? `throw new Error( "Unexpected line: " + line );` Do we need a new class in the `THREE` namespace just for this?

**Comment by vidstige** [comment]
Alright, this diff should be rather lean and mean. What do you think about this @mrdoob ?

**Comment by mrdoob** [comment]
Perfect!

**Comment by mrdoob** [comment]
Thanks! 😊

**Comment by vidstige** [comment]
Thank _you_ for feedback & merge. :+1:

**Pull request merged**